### ISSUE TITLE
Many to many support

### ISSUE THREAD

**Issue by AndreasBackx** [Raizlabs/DBFlow#264]
In #28 you said that there was no many-to-many support yet. Are there any plans to implement many to many into DBFlow by automatically generating a table between the BaseModels? Currently it seems that you'll have to create the m2m table yourself or is there a better way?

**Comment by AndreasBackx** [comment]
I'm having a problem when trying to make the 2 ForeignKeys in the M2M table both primary keys.

I'm using the following model to handle M2M relations:
StoreCategoryMany.java
```java
@Table
public class StoreCategoryMany extends BaseModel {

	@Column
	@PrimaryKey
	@ForeignKey(references = {@ForeignKeyReference(columnName = "store", columnType = long.class, foreignColumnName = "id")},
			saveForeignKeyModel = false,
			onDelete = ForeignKeyAction.CASCADE,
			onUpdate = ForeignKeyAction.CASCADE)
	protected ForeignKeyContainer<Store> store;

	@Column
	@PrimaryKey
	@ForeignKey(references = {@ForeignKeyReference(columnName = "category", columnType = long.class, foreignColumnName = "id")},
			saveForeignKeyModel = false,
			onDelete = ForeignKeyAction.CASCADE,
			onUpdate = ForeignKeyAction.CASCADE)
	protected ForeignKeyContainer<StoreCategory> category;

	public StoreCategoryMany() {

	}

	public StoreCategoryMany(Store store, StoreCategory category) {
		this.store.setModel(store);
		this.category.setModel(category);
	}
}
```

When I build the application, I get errors in StoreCategoryMany$Adapter stating StoreCategoryMany$Table.STORE and StoreCategoryMany$Table.CATEGORY do not exist (that's true, it's STORE_STORE and CATEGORY_CATEGORY).

StoreCategoryMany$Adapter.java
```java
  @Override
  public ConditionQueryBuilder<StoreCategoryMany> getPrimaryModelWhere(StoreCategoryMany model) {
    return new ConditionQueryBuilder<StoreCategoryMany>(StoreCategoryMany.class, Condition.column(StoreCategoryMany$Table.STORE).is((model.store.getValue("store"))),Condition.column(StoreCategoryMany$Table.CATEGORY).is((model.category.getValue("category"))));
  }

  @Override
  public ConditionQueryBuilder<StoreCategoryMany> createPrimaryModelWhere() {
    return new ConditionQueryBuilder<StoreCategoryMany>(StoreCategoryMany.class, Condition.column(StoreCategoryMany$Table.STORE).is("?"),Condition.column(StoreCategoryMany$Table.CATEGORY).is("?"));
  }
```

**Comment by agrosner** [comment]
Kind of strange that you make the `@PrimaryKey` and `@ForeignKey` objects the same together. I'd recommend creating an `@PrimaryKey(autoincrement=true)` field that becomes the primary key for that table. I think the processor is getting confused since you've defined the columns as both.

**Comment by mathieudebrito** [comment]
@agrosner Creating an auto-incremented field here would not be very "db relation oriented"

**Comment by agrosner** [comment]
its fixed in 3.0.0 with the `@ManyToMany` annotation, as well as auto-gen of that table you have to create and support for primary as foreign keys.
